File: 2-Patron/python/Armadura.py
class Armadura(object):

    def __init__(self, nombre, nivel, danio_recibido, duracion):
        self.__nombre = nombre
        self.__nivel = nivel
        self.__danio_recibido = danio_recibido
        self.__duracion = duracion

    @property
    def nombre(self):
        return self.__nombre

    @nombre.setter
    def nombre(self, new_nombre):
        self.__nombre = new_nombre

    @property
    def nivel(self):
        return self.__nivel

    @nivel.setter
    def nivel(self, new_nivel):
        self.__nivel = new_nivel

    @property
    def danio_recibido(self):
        return self.__danio_recibido

    @danio_recibido.setter
    def danio_recibido(self, new_danio_recibido):
        self.__danio_recibido = new_danio_recibido

    @property
    def duracion(self):
        return self.__duracion

    @duracion.setter
    def duracion(self, new_duracion):
        self.__duracion = new_duracion

    def incDañoRecibido(self, incremento):
        self.__danio_recibido += incremento

    def incDuracion(self, incremento):
        self.__duracion += incremento

    def incNivel(self):
        self.__nivel += 1

File: 1-Patron/php/Armadura.php
<?php
class Armadura
{
  private $nombre;
  private $nivel;
  private $dano_recibido;
  private $duracion;

  function __construct($nombre, $nivel, $dano_recibido, $duracion)
  {
    $this->nombre = $nombre;
    $this->nivel = $nivel;
    $this->dano_recibido = $dano_recibido;
    $this->duracion = $duracion;
  }

  function getNombre()
  {
    return $this->nombre;
  }

  function setNombre($nombre)
  {
    $this->nombre = $nombre;
  }

  function getNivel()
  {
    return $this->nivel;
  }

  function setNivel($nivel)
  {
    $this->nivel = $nivel;
  }

  function getDano_Recibido()
  {
    return $this->dano_recibido;
  }

  function setNivel($dano_recibido)
  {
    $this->dano_recibido = $dano_recibido;
  }

  function getDuracion()
  {
    return $this->duracion;
  }

  function setDuracion($duracion)
  {
    $this->duracion = $duracion;
  }

  function incDanoRecibido($incremento)
  {
    $this->dano_recibido += $incremento;
  }

  function incDuracion($incremento)
  {
    $this->duracion += $incremento;
  }

  function incNivel()
  {
    $this->nivel += 1;
  }
}
?>
